src/validate.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import chalk from "chalk";
+import * as fs from "fs";
+import * as path from "path";
+import { CliError } from "./cli-error";
+
+export class Validate {
+
+    constructor(readonly pluginRootFolder: string) {
+
+    }
+
+    public async check(): Promise<boolean> {
+        // load package.json
+        const packageJsonPath = this.pluginRootFolder + path.sep + "package.json";
+
+        if (!fs.existsSync(packageJsonPath)) {
+            throw new CliError("The plugin is not having any package.json file");
+        }
+        const pluginPackageJson: any = require(packageJsonPath);
+        this.validateEngine(pluginPackageJson, packageJsonPath);
+        this.validatePluginEntry(pluginPackageJson, packageJsonPath);
+
+        // TODO: check for yarn.lock
+
+        return Promise.resolve(true);
+    }
+
+    public validateEngine(pluginPackageJson: any, packageJsonPath: string): void {
+
+        const missingEngineSuggest = "You need to add "
+            + chalk.italic('\n  "engines": \{\n    "theia": "latest"\n  \}\n')

```

Comment:
I suppose here should be `"`**`theiaPlugin`**`": "latest"` instead of `"`**`theia`**`": "latest"`

Reply:
I was thinking it was the name of the product (theia) but if you think it's better to say it's the name of the component I'm fine as well.

Reply:
I think engine name should be related to the theia's plugin system but not to theia itself. 
WDYT?

Reply:
I will update the code, maybe there will other changes before it goes public when other will review, but for now let's make it to theia plugin's system version

Reply:
fixed